pkgs/development/python-modules/shapely/default.nix

```diff
@@ -22,11 +22,16 @@ buildPythonPackage rec {
     export LANG="en_US.UTF-8";
   '';
 
-  patchPhase = let
+  postPatch =  let
     libc = if stdenv.isDarwin then "libc.dylib" else "libc.so.6";
   in ''
-    sed -i "s|_lgeos = load_dll('geos_c', fallbacks=.*)|_lgeos = load_dll('geos_c', fallbacks=['${geos}/lib/libgeos_c${stdenv.hostPlatform.extensions.sharedLibrary}'])|" shapely/geos.py
-    sed -i "s|free = load_dll('c').free|free = load_dll('c', fallbacks=['${stdenv.cc.libc}/lib/${libc}']).free|" shapely/geos.py
+    substituteInPlace shapely/geos.py \
+      --replace "_lgeos = load_dll('geos_c', fallbacks=alt_paths)" "_lgeos = load_dll('geos_c', fallbacks=['${geos}/lib/libgeos_c${stdenv.hostPlatform.extensions.sharedLibrary}'])" \
+      --replace "free = load_dll('c').free" "free = load_dll('c', fallbacks=['${stdenv.cc.libc}/lib/${libc}']).free"
+  '';
+
+  preBuild = ''
+    export GEOS_LIBRARY_PATH="${geos}/lib/libgeos_c${stdenv.hostPlatform.extensions.sharedLibrary}"
```

Comment:
Why is this needed in addition to the patch?

Reply:
fixed. found a way to do in the path phase